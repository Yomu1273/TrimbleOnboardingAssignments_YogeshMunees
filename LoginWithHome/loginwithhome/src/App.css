.App {
  text-align: center;
}

.Heading {
  background-color:rgb(211, 211, 211);
  margin:0;
  top:0;
  left:0;
  color:rgb(8, 67, 155);
  text-align: center;
  display: flex;
  justify-content: center;
  text-align: center;
}

#TrimbleLogo {
  position: absolute;
  right: 15px;
  top: 15px;
}

.box {
  margin-top: 2%;
  margin-left: 15%;
  margin-right: 15%;
  border: 2px solid black;
  border-radius: 15px;
}

.modeButtons {
  padding-top: 2%;
}

.login{
  background-color: rgb(8, 67, 155);
  color: white;
  border: none;
  padding: 1%;
  padding-bottom: 2%;
  width: 25%;
  border-radius: 5px 5px 0 0;
}

.signup {
  background-color: rgb(8, 67, 155);
  color: white;
  border: none;
  padding: 1%;
  width: 25%;
  border-radius: 5px 5px 0 0;
}

.modeFields {
  padding: 2%;
  background-color: rgb(8, 67, 155);
  color: white;
  display: flex;
}

.navbar {
  list-style: none;
  display: flex;
  justify-content: center;
  text-align: center;
}

.listitem {
  padding: 2%;
}

.logout {
  background-color: white;
  border: 0px solid black;
  color: red;
}