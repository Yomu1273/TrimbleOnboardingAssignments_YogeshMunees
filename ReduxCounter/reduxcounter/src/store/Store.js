import {configureStore} from "@reduxjs/toolkit";
import counterDancer from "../slices/CounterSlice";

export const store = configureStore({
    reducer:{
        counter: counterDancer,
    }
});