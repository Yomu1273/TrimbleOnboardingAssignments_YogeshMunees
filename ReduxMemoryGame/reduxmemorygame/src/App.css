.App {
  text-align: center;
}

.doors {
  display: flex;
  justify-content: center;
  text-align: center;
  padding: 2%;
}

.door {
  border: 2px solid black;
  padding: 5%;
}

.door:hover {
  background-color: grey;
}