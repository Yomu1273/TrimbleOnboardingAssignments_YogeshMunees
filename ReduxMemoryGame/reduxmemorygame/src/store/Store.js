import {configureStore} from '@reduxjs/toolkit';
import memorySliceReducer from '../slice/memorySlice';

export const store = configureStore({
    reducer:{
        memory:memorySliceReducer,
    }
});