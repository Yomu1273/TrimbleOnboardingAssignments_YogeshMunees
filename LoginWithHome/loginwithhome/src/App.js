import './App.css';
import {BrowserRouter, Routes, Route, Link} from 'react-router-dom';
import NavPage from './pages/NavPage';
import HomePage from './pages/HomePage';
import ContactPage from './pages/ContactPage';
import Authenticator from './Authenticator';
import { isAuthenticated,setCurrentLogger,resetCurrentLogger } from './slices/LoginSlice';
import {useSelector} from 'react-redux';

function App() {

  const canweallow = useSelector((state)=>state.login.currentLogger);

  return (
    <>
      <div className='Heading'>
        <h1>Dream Bell</h1>
        <img id="TrimbleLogo" src="https://companieslogo.com/img/orig/TRMB-f6434f42.png" width="50px" height="50px"/>
      </div>
      <BrowserRouter>
          <Routes>
            <Route element={<Authenticator canweallow={canweallow}/>}>
              <Route path="/" element={<NavPage/>}>
              <Route index element={<HomePage/>}/>
              <Route path='contact' element={<ContactPage/>}/>
              </Route>
            </Route>
          </Routes>
        </BrowserRouter>
    </>
  );
}

export default App;