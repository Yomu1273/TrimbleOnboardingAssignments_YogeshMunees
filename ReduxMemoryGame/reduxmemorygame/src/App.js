import { Provider } from 'react-redux';
import './App.css';
import Memory from './Memory';

function App() {
  return (
    <div className="App">
      <h1>Memory Test using Redux</h1>
      <div>
          <Memory/>
      </div>
    </div>
  );
}

export default App;
