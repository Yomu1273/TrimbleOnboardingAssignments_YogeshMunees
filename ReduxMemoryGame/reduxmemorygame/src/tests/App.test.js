import React from 'react';
import {render,cleanup} from '@testing-library/react';
import App from '../App';
import { Provider } from 'react-redux';
import {store} from '../store/Store';

afterEach(cleanup);

it('should take a snapshot',()=>{
    const {asFragment} = render(<Provider store={store}><App/></Provider>)
    expect(asFragment(<App/>)).toMatchSnapshot()
})