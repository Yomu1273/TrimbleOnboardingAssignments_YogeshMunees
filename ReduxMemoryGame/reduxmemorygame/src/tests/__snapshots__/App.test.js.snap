// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Memory Test using Redux
    </h1>
    <div>
      <div
        data-testid="doorDivision"
        style="visibility: hidden;"
      >
        <div>
          <h3>
            Door Step 1
          </h3>
        </div>
        <div
          class="doors"
        >
          <div
            class="door"
            data-testid="leftDoor"
          >
            |
          </div>
          <div
            style="visibility: hidden;"
          >
            gapbetweenthedoors
          </div>
          <div
            class="door"
            data-testid="rightDoor"
          >
            |
          </div>
        </div>
      </div>
      <div>
        <button
          data-testid="generateQuestionBtn"
        >
          Generate Question
        </button>
        <h3
          data-testid="questionSentence"
          style="visibility: hidden;"
        >
          Your sequence is: 
        </h3>
        <h3
          data-testid="memorizeInstruction"
          style="visibility: hidden;"
        >
          Memorize the Sequence
        </h3>
        <button
          data-testid="goBtn"
          style="visibility: hidden;"
        >
          GO
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
